main.cpp

```diff
@@ -31,6 +31,11 @@
 #pragma warning(disable : 4996)
 #endif
 
+#ifdef __clang__
+#pragma clang diagnostic ignored "-Wswitch-enum"
```

Comment:
Yes, you are better to add support building SPIRV-Cross with clang and gcc in Travis build, and add "-Weverything -Werror -Wno-c++98-compat-pedantic -std=c++11" when building with clang. My example in tinyobjloader may help  to setup compiler matrix:  https://github.com/syoyo/tinyobjloader/blob/master/.travis.yml